### PR TITLE
sway-overfocus: init at 0.2.3-fix

### PR DESCRIPTION
**Pull request by ivan770**
## Description of changes

`sway-overfocus` is a tool for i3 and Sway-based desktops that provides capabilities for a more precise window/container focus control.

Homepage: https://github.com/korreman/sway-overfocus

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/sw/sway-overfocus/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  fetchFromGitHub,
+  lib,
+  nix-update-script,
+  rustPlatform,
+  ...
```
Comment (WolfangAukang) [inline pkgs/by-name/sw/sway-overfocus/package.nix:6]: Please do not use elipsis on packages. Pull what is needed.

File: pkgs/by-name/sw/sway-overfocus/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "sway-overfocus";
+  version = "0.2.3-fix";
+
+  src = fetchFromGitHub {
+    owner = "korreman";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-KHbYlxgrrZdNKJ7R9iVflbbP1c6qohM/NHBSYuzxEt4=";
```
Comment (WolfangAukang) [inline pkgs/by-name/sw/sway-overfocus/package.nix:17]:
```suggestion
    hash = "sha256-KHbYlxgrrZdNKJ7R9iVflbbP1c6qohM/NHBSYuzxEt4=";
```

File: pkgs/by-name/sw/sway-overfocus/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  cargoSha256 = "sha256-zp6PSu8P+ZUhrqi5Vxpe+z9zBaSkdVQBMGNP0FVOviQ=";
+
+  # Crate without tests.
+  doCheck = false;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
```
Comment (WolfangAukang) [inline pkgs/by-name/sw/sway-overfocus/package.nix:27]:
```
   changelog = "https://github.com/korreman/sway-overfocus/releases/tag/${src.rev}"
```

Comment (WolfangAukang) [inline pkgs/by-name/sw/sway-overfocus/package.nix:27]: Missing a `mainProgram` on the `meta` section
```
[nix-shell:~/.cache/nixpkgs-review/pr-267765]$ ./results/sway-overfocus/bin/sway-overfocus
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]

**Comment by WolfangAukang** [comment]
Please create a separate commit when adding yourself to the maintainer list

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 267765` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sway-overfocus</li>
  </ul>
</details>

**Review by WolfangAukang — APPROVED** [review]

File: pkgs/by-name/sw/sway-overfocus/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.2.3-fix";
+
+  src = fetchFromGitHub {
+    owner = "korreman";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-KHbYlxgrrZdNKJ7R9iVflbbP1c6qohM/NHBSYuzxEt4=";
+  };
+
+  cargoSha256 = "sha256-zp6PSu8P+ZUhrqi5Vxpe+z9zBaSkdVQBMGNP0FVOviQ=";
```
Comment (Lurkki14) [inline pkgs/by-name/sw/sway-overfocus/package.nix:14]:
```suggestion
  cargoHash = "sha256-zp6PSu8P+ZUhrqi5Vxpe+z9zBaSkdVQBMGNP0FVOviQ=";
```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Review by Lurkki14 — APPROVED** [review]

**Pull request merged**